### PR TITLE
ctrl+p now previews the curva as the default instruction.

### PR DESCRIPTION
**Pull request by atlas0fd00m**

### CODE REVIEW

File: vivisect/qt/memory.py [hunk 1]
```diff
@@ -56,6 +56,8 @@ def __init__(self, *args, **kwargs):
         self.addHotKey('ctrl+S', 'viv:make:struct:again')
         self.addHotKey('ctrl+meta+S', 'viv:make:struct:multi')
         self.addHotKey('U', 'viv:undefine')
+        self.addHotKey('ctrl+p', 'viv:preview:instr')
+        self.addHotKey('ctrl+s', 'viv:save')
```
Comment (rakuy0) [inline vivisect/qt/memory.py]: Would the `ctrl+s` here conflict with the `ctrl+s` that the main window hooks (ie, would one preempt the other and make it so the other is not called)?

File: vivisect/qt/memory.py [hunk 2]
```diff
@@ -89,7 +91,7 @@ def _applyColorMap(self, cmap):
 
     @vq_hotkey.hotkey('viv:nav:nextva')
     def _hotkey_nav_nextva(self):
-        if self._canv_curva == None:
+        if self._canv_curva is None:
```
Comment (rakuy0) [inline vivisect/qt/memory.py]: :D

File: vivisect/qt/memory.py [hunk 3]
```diff
@@ -257,29 +259,41 @@ def makeStructAgainMulti(self, va, parent=None):
...
             self.vw.makeNumber(self._canv_curva, 16)
 
+    @vq_hotkey.hotkey('viv:preview:instr')
+    def _hotkey_preview_instr(self):
+        if self._canv_curva is not None:
+            self.vw.previewCode(self._canv_curva)
+
+    @firethread
+    @vq_hotkey.hotkey('viv:save')
+    def _hotkey_save(self, fullsave=False):
```
Comment (rakuy0) [inline vivisect/qt/memory.py]: So there's also `_menuFileSave` in the main qt window that does exactly this. This smells like something that should be carved out to a common base class (like the HotKeyMixin class).

Comment (atlas0fd00m) [reply vivisect/qt/memory.py]: good catch!  removed.  this code apparently predates when we wrapped ctrl-s saving into mainline :)

File: vivisect/qt/memory.py (hunk 1 above)
Comment (atlas0fd00m) [reply vivisect/qt/memory.py]: they're the same thing.  removed. (see note on other)

**Comment by atlas0fd00m** [comment]
maybe a specialized hotkey handler?  
some hotkeys don't need a valid canv_curva (like ctrl-s) ;)
but if we did make a specialty decorator for those that require canv_curva to be set, that would save a lot of code.

**Comment by atlas0fd00m** [comment]
the merge seems to have taken place... although for some reason this PR doesn't show as complete.  closing.

**Pull request closed without merging**